adal/oauth2_client.py

```diff
@@ -138,7 +138,7 @@ def _parse_id_token(self, encoded_token):
                 self._log.warn('The returned id_token could not be base64 url safe decoded.')
                 return
 
-            id_token = json.loads(b64_decoded.decode())
+            id_token = json.loads(b64_decoded.decode('utf-8'))

```

Comment:
You know, this is a breaking change, but  figure you are OK with. I have not seen other failures on decoding, so let us go ahead